### PR TITLE
Save greyed out when no project loaded; `Menu` updates `QAction` to saved state on `set_action`

### PR DESCRIPTION
**Pull request by zwimer**
Fixes: https://github.com/angr/angr-management/issues/653

### CODE REVIEW

File: angrmanagement/data/instance.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ class Instance:
     """
     An object to give access to normal angr project objects like a Project, CFG, and other analyses.
     """
-    project: Union[angr.Project, ObjectContainer]
+    project: ObjectContainer
```
Comment (mborgerson) [inline angrmanagement/data/instance.py:40]: Presumably the annotations were like this for code completion. We should be consistent in style though; that is, either switch all of these lines over to annotate only ObjectContainer, or do the Union with the real type for completion. I'll defer to @ltfish on it

File: angrmanagement/ui/menus/file_menu.py [hunk 2]
```diff
@@ -45,6 +52,11 @@ def __init__(self, main_window):
             MenuEntry('E&xit', main_window.quit),
         ])
 
+    def _edit_save(self, **_):
+        enable: bool = not self._project.am_none
+        for i in self._save_entries:
+            (i.enable if enable else i.disable)()
```
Comment (mborgerson) [inline angrmanagement/ui/menus/file_menu.py:58]: nit: Simpler to just pass the bool in to a setEnabled function

Comment (zwimer) [reply angrmanagement/ui/menus/file_menu.py:58]: There is no `setEnabled` function; we are editing from the custom class which exposes only these two functions- correct me if I am wrong.

Comment (zwimer) [reply angrmanagement/ui/menus/file_menu.py:58]: https://github.com/angr/angr-management/blob/6c70c966e52ffc99e5d609c3e5261888f4bc5a3c/angrmanagement/ui/menus/menu.py#L5

**Pull request merged**